### PR TITLE
dev/core#1471 - Add system status alert for deleted custom fields in …

### PR DESCRIPTION
**Pull request by jitendrapurohit**
…smart groups

Overview
----------------------------------------
Add message alert on system status page for deleted custom fields on smart group.

Before
----------------------------------------
See https://lab.civicrm.org/dev/core/issues/1471. There is no indication or way to know which smart groups are responsible for errors on the group page.

After
----------------------------------------
System status show an alert with a list of groups responsible for smart group errors.

image

UPDATED text after QA -

image

Comments
----------------------------------------
Gitlab - https://lab.civicrm.org/dev/core/issues/1471

### CODE REVIEW

**Comment by francescbassas** [comment]
In the alert message, instead of deleting problematic smart groups, I might suggest editing their search criteria and updating them to clean outdated fields from saved search.

**Comment by mattwire** [comment]
test this please

**Comment by jitendrapurohit** [comment]
@francescbassas updated the msg to suggest editing and updating the search criteria. Also updated the table to also list custom field labels of deleted fields that are included in the saved search form values.

**Comment by francescbassas** [comment]
Great, looks better!

Custom field value isn't displayed in my case. Maybe because they are deleted and not disabled?

screenshot

I would also add a :pencil2: to each row to link to the group search criteria `.../civicrm/contact/search/advanced?reset=1&ssID=5` for easy editing.

**Comment by yashodha** [comment]
@jitendrapurohit @francescbassas Edit/settings button sound good. Can the list be group title which is link to the results as well.

**Comment by francescbassas** [comment]
In the column **Custom field** I would put:

- Deleted (red color) - *if field has been removed*
- The name of the field (disabled color with a link to edit group `...civicrm/admin/custom/group/field/update?action=update&reset=1&gid=XX&id=XX`) - *if field has been disabled*

**Comment by yashodha** [comment]
@francescbassas sounds good. @jitendrapurohit  Can you please make the requested changes? Is the test failure related?

**Comment by eileenmcnaughton** [comment]
I don't think the test failure is related - @yashodha go  ahead & merge this once you are happy with  it

**Comment by eileenmcnaughton** [comment]
test this  please

**Comment by jitendrapurohit** [comment]
I've updated the PR with the additional requirements. The updated SS is as shown below -

image

File: CRM/Utils/Check/Component/Schema.php [hunk 1]
```diff
@@ -88,4 +88,76 @@ public function checkMissingLogTables() {
...
+              ),
+              2 => $customField['label'],
+            ]);
+          }
+          catch (Exception $e) {
+            $fieldName = ' <span style="color:red"> - Deleted - </span> ';
+          }
+        }
+
+        $html .= "<tr><td>{$id}</td><td>{$field['title']}</td><td class='disabled'>{$fieldName}</td>";
```
Comment (francescbassas) [inline CRM/Utils/Check/Component/Schema.php]: To add edit and config group icons:

```
        $groupEdit = '<a href="' . CRM_Utils_System::url('civicrm/contact/search/advanced', "?reset=1&ssID={$id}", TRUE) . '" title="' . ts('Edit search criteria') . '"> <i class="crm-i fa-pencil"></i> </a>';
        $groupConfig = '<a href="' . CRM_Utils_System::url('civicrm/group', "?reset=1&action=update&id={$id}", TRUE) . '" title="' . ts('Group settings') . '"> <i class="crm-i fa-gear"></i> </a>';
        $html .= "<tr><td>{$id} {$groupEdit} {$groupConfig}</td><td>{$field['title']}</td><td class='disabled'>{$fieldName}</td>";
```
Aso I would convert the 'ID' and 'Group Title' and gr to one 'Group'

proposal

Comment (francescbassas) [reply CRM/Utils/Check/Component/Schema.php]: Note that I put on $groupConfig the group ID when should be the SSID.

**Comment by jitendrapurohit** [comment]
@francescbassas The above change has been made.  Can you pls retest and confirm? Thanks.

**Comment by jitendrapurohit** [comment]
test this please

**Comment by francescbassas** [comment]
Tested and looks good. Thanks @jitendrapurohit.

final

Looking again, I could just only up the last paragraph above the table.

test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@francescbassas is this good to merge once tests pass?

**Comment by jitendrapurohit** [comment]
>Looking again, I could just only up the last paragraph above the table.

done @francescbassas

**Comment by yashodha** [comment]
@jitendrapurohit Can you please update the latest screenshot in After section in which all the changes have been incorporated or is this the one https://user-images.githubusercontent.com/1064675/72603392-37b81d00-3919-11ea-8d93-1bc121f6be74.png?

**Comment by jitendrapurohit** [comment]
@yashodha Done.

**Comment by yashodha** [comment]
@jitendrapurohit looks good, merging this. Thanks!

**Comment by eileenmcnaughton** [comment]
nice! Thanks  all

**Pull request merged**